### PR TITLE
--source-image should count as a source input

### LINKED ISSUE #17675: oc new-build: can't figure out how to specify image stream to overwrite FROM
I'm trying to create a `Build` that will take in source from one image and use another image as the build FROM. I've got this:
```
$ oc new-build --source-image pr-21-257717b093c5e0b975dd6042:src                                        \
               --source-image-path /data/src/github.com/openshift/image-registry/images/dockerregistry/ \
               --image-stream pr-21-257717b093c5e0b975dd6042:origin-base                                \
               --to pr-21-257717b093c5e0b975dd6042:origin-docker-registry                               \
               --output json                                                                            \
               --dry-run
error: the image match "ci-openshift-ansible/pr-21-257717b093c5e0b975dd6042:origin-base" for source repository "pr-21-257717b093c5e0b975dd6042:src" does not appear to be a source-to-image builder.

- to attempt to use this image as a source builder, pass "--strategy=source"
- to use it as a base image for a Docker build, pass "--strategy=docker"
```

So I add the `--strategy=docker` and I get:

```
$ oc new-build --source-image pr-21-257717b093c5e0b975dd6042:src                                        \
               --source-image-path /data/src/github.com/openshift/image-registry/images/dockerregistry/ \
               --image-stream pr-21-257717b093c5e0b975dd6042:origin-base                                \
               --strategy docker                                                                        \
               --to pr-21-257717b093c5e0b975dd6042:origin-docker-registry                               \
               --output json                                                                            \
               --dry-run
error: --strategy is specified and none of the arguments provided could be classified as a source code location
```

This seems wonky. 

/kind bug
/cc @bparees @csrwng

### PR DESCRIPTION
**Pull request by coreydaley**
- the --source-image flag should count as a source input
 - the --strategy flag should override the build strategy in all source repositories

Fixes #17675

### CODE REVIEW

**Comment by coreydaley** [comment]
@openshift/sig-developer-experience ptal
if this makes sense I will figure out some tests around the change

**Comment by bparees** [comment]
not seeing how this would fix `--binary is ignoring the --strategy=docker flag (it created an s2i strategy build)` ?

but the reordering makes sense for the `providing an image input should count for purposes of supplying "source" (you shouldn't have to pass --binary to trick oc new-build into creating the build w/o providing an explicit source repo)` part of the issue.

**Comment by coreydaley** [comment]
@bparees ptal

File: test/cmd/newapp.sh [hunk 1]
```diff
@@ -469,6 +469,12 @@ os::cmd::expect_success 'oc new-app openshift/bogusimage https://github.com/open
 
 os::cmd::expect_success 'oc create -f test/testdata/installable-stream.yaml'
 
+# ensure that --strategy sets the build strategy
+os::cmd::expect_success_and_text 'oc new-build --name sourcetest python~https://github.com/sclorg/django-ex --source-image centos:latest --source-image-path /tmp --strategy source --dry-run -o yaml' 'sourceStrategy'
+os::cmd::expect_success_and_text 'oc new-build --name sourcetest python~https://github.com/sclorg/django-ex --source-image centos:latest --source-image-path /tmp --strategy pipeline --dry-run -o yaml' 'jenkinsPipelineStrategy'
+os::cmd::expect_success_and_text 'oc new-build --name sourcetest python~https://github.com/sclorg/django-ex --source-image centos:latest --source-image-path /tmp --strategy docker --dry-run -o yaml' 'dockerStrategy'
```
Comment (bparees) [inline test/cmd/newapp.sh]: can you also add a test that confirms you can create a build with just image inputs (w/o passing the --binary flag)?

Comment (coreydaley) [reply test/cmd/newapp.sh]: updated

**Comment by coreydaley** [comment]
@bparees ptal

**Pull request merged**